### PR TITLE
Feature/reanimation

### PR DESCRIPTION
**Pull request by sebastianfdez**

### CODE REVIEW

File: src/components/shared/Reanimate.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<style>
+.button-container {
+  width: 100%;
+  max-width: 300px;
+}
+.button {
+  min-width: fit-content !important;
+}
+</style>
```
Comment (manuelaguirre) [inline src/components/shared/Reanimate.vue:36]: Given this component hardly presents any logic, we could get it to be a generic button.

File: src/components/shared/InsertInput.vue [hunk 2]
```diff
@@ -32,7 +31,7 @@ export default {
   },
   methods: {
     insertValue() {
-      if (this.insertedValue !== '') {
+      if (this.insertedValue !== '' && !this.disabled) {
```
Comment (manuelaguirre) [inline src/components/shared/InsertInput.vue:34]: Nice

File: src/components/BTreeVisualizer.vue [hunk 3]
```diff
@@ -162,6 +169,23 @@ export default {
...
+      this.isAnimating = true;
+      for (let i = 1; i < frames.length; i += 1) {
+        setTimeout(() => {
+          this.currentSequence.addFrame(frames[i]);
+          if (i === frames.length - 1) {
+            this.isAnimating = false;
+          }
+          this.currentFrame += 1;
+          this.sequencesList = Object.assign(this.sequencesList);
+        }, i * 500);
```
Comment (manuelaguirre) [inline src/components/BTreeVisualizer.vue:186]: we should stop hardcoding this number and assign it to some value of the instance (we can take it up later to dynamically change the speed)

Comment (manuelaguirre) [reply src/components/BTreeVisualizer.vue:186]: also, was it possible to reuse the `addSequenceAsync` method?

Comment (sebastianfdez) [reply src/components/BTreeVisualizer.vue]: TODO

**Pull request merged**